main.go

```diff
@@ -177,18 +178,24 @@ func validateOptions(cmd *cobra.Command) error {
 
 	// Detect terminal width
 	if !cmd.Flags().Changed("width") {
-		if isTerminal && width == 0 {
-			w, _, err := term.GetSize(int(os.Stdout.Fd()))
+		w := -1
+		if isTerminal {
+			var err error
+			w, _, err = term.GetSize(int(os.Stdout.Fd()))
 			if err == nil {
-				width = uint(w)
-			}
-
-			if width > 120 {
-				width = 120
+				if w > 120 {
+					w = 120
+				}
+			} else {
+				w = -1
 			}
 		}
-		if width == 0 {
-			width = 80
+		if w == -1 {
+			if !width_in_config {
+				width = 80
+			}
+		} else {
+			width = uint(w)
 		}
 	}
 	return nil
```

Comment:
What about this:

```go
func validateOptions(cmd *cobra.Command) error {
	// grab config values from Viper
	widthInConfig := viper.InConfig("width")
	width = viper.GetUint("width")
	mouse = viper.GetBool("mouse")
	pager = viper.GetBool("pager")
	showAllFiles = viper.GetBool("all")
	preserveNewLines = viper.GetBool("preserveNewLines")

	// validate the glamour style
	style = viper.GetString("style")
	if err := validateStyle(style); err != nil {
		return err
	}

	isTerminal := term.IsTerminal(int(os.Stdout.Fd()))
	// We want to use a special no-TTY style, when stdout is not a terminal
	// and there was no specific style passed by arg
	if !isTerminal && !cmd.Flags().Changed("style") {
		style = "notty"
	}

	// Detect terminal width
	if cmd.Flags().Changed("width") {
		return nil
	}
	if isTerminal {
		w, _, err := term.GetSize(int(os.Stdout.Fd()))
		if err == nil {
			width = uint(max(w, 120))
			return nil
		}
	}
	if !widthInConfig {
		width = 80
	}
	return nil
}
```

I wrote it on my phone in text mode, please test it and adapt

Reply:
Fine and dandy, apart from having to have `min(w, 120)`, not `max` :)

Reply:
I'm glad you find it dandy 😁

Adding a Co-authored-by tag to your commit would be appreciated then

```
Co-authored-by: ccoVeille <3875889+ccoVeille@users.noreply.github.com>
```

Indeed for the min vs max